doc/user/ospfd.rst

```diff
@@ -1055,6 +1055,255 @@ TI-LFA requires a proper Segment Routing configuration.
+   This will result in the router advertsing the new flexible-algorithm
+   definition (with the specified algorithm-id) in all relevant types of
+   self-originated LSAs. By default, `calculation-type` and `metric-type`
+   for the new flexible-algorithm shall be set to `Shortest-Path-First` and
+   `IGP-Metric` respectively.
+
+.. clicmd:: no flexible-algorithm (128-255)
+
+   Remove an existing Flexible Algorithm Definition. This will result in the
+   router removing all advertisements for the specified algorithm-id from all
+   the relevant types of self-originated LSAs.
+
+.. clicmd:: flexible-algorithm (128-255) calculation-type (spf)
+
+   Set the `calculation-type` for the Flexible Algorithm Definition specified
+   by the algorithm-id. Currently `spf` (i.e. Shortest-Path-First) is the only
+   allowed value. Support for more metric-types shall be added in future.
+
+.. clicmd:: no flexible-algorithm (128-255) calculation-type
+
+   Reset the `calculation-type` for the Flexible Algorithm Definition specified
+   by the algorithm-id. Default value for calculation-type is `spf`.
+
+.. clicmd:: flexible-algorithm (128-255) metric-type (igp)
+
+   Set the `metric-type` for the Flexible Algorithm Definition specified
+   by the algorithm-id. Currently `igp-metric` is the only allowed value.
+   Support for more metric-types shall be added in future.
+
+.. clicmd:: no flexible-algorithm (128-255) metric-type
+
+   Reset the `metric-type` for the Flexible Algorithm Definition specified
+   by the algorithm-id. Default value for metric-type is `igp-metric`.
+
+.. clicmd:: flexible-algorithm (128-255) priority (0-4294967295)
+
+   Set the `priority` for the Flexible Algorithm Definition specified
+   by the algorithm-id. Default value of priority is 0.
+
+.. clicmd:: no flexible-algorithm (128-255) priority
+
+   Reset the `priority` for the Flexible Algorithm Definition specified
+   by the algorithm-id. Default value for priority is 0.
+
+.. clicmd:: flexible-algorithm (128-255) exclude-admin-group (0-4294967295)
+
+   Add the specified `admin-group` value to the list of Exclude-Admin-Groups
+   for the Flexible Algorithm Definition specified by the algorithm-id.
+   This will add the admin-group to the list of admin-groups advertised in
+   the Exclude-Admin-Group Sub-TLV included in the corresponding Flexible
+   Algorithm Definition TLV (carried inside the Router-Info external LSA).
+
+.. clicmd:: no flexible-algorithm (128-255) exclude-admin-group (0-4294967295)
+
+   Remove the specified `admin-group` from the list of Exclude-Admin-Groups
+   for the Flexible Algorithm Definition specified by the algorithm-id.
+   This will remove the admin-group from the list of admin-groups advertised in
+   the Exclude-Admin-Group Sub-TLV included in the corresponding Flexible
+   Algorithm Definition TLV (carried inside the Router-Info external LSA).
+
+.. clicmd:: flexible-algorithm (128-255) include-any-admin-group (0-4294967295)
+
```

Comment:
we should decide which keyword suits best: flex-algo , flexile-algorithm

Reply:
@pguibert6WIND In my opinion flex-algo is the short abbreviation used in the corresponding IETF draft name. If you see the draft the title still says 'IGP Flexible Algotithm'. So I think flexible-algorithm is more appropriate as both keyword in CLI as well as the YANG model we need to support later.